### DISCUSSION TITLE
Is creating presigned POST to upload a file supported?

### DISCUSSION THREAD

**Discussion by rossgray** [minio/minio#18689] [Q&A]
I was previously using a presigned POST request to generate presigned URLs in order to let users upload files via POST requests, via the boto3 `generate_presigned_post` method: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-presigned-urls.html#generating-a-presigned-url-to-upload-a-file

This was working fine until a recent release of MinIO (I think [this PR](https://github.com/minio/minio/commit/6dec60b6e6fe912f5e4b63062fa8d4a2ac1e1b45) stopped this working. I am getting an error when POSTing the file to the presigned URL: `Access Denied. (Each form field that you specify in a form (except Signature, Awsaccesskeyid) must appear in the list of conditions.)` even though these form fields were returned by MinIO when calling `generate_presigned_post()`.

I don't see this functionality mentioned in the MinIO docs anywhere so am wondering whether this functionality is actually supported and whether I need to use a presigned PUT request instead?

**Comment by klauspost** [comment]
The signature is generated by the SDK you use. Check if you are using the latest version - and tell us what it ism if it is.

**Comment by rossgray** [comment]
When you say SDK, do you mean `boto3` or MinIO? I'm not using a MinIO SDK

**Comment by rossgray** [comment]
I'm running MinIO in Docker. The version is `minio version RELEASE.2023-12-20T01-00-02Z (commit-id=7a311a3b666a0e5476fbde24d4413220876d9df2)`

**Comment by rossgray** [comment]
@klauspost could you let me know if presigned POST requests are supposed to work? If so, I can try to create an example to reproduce the issue I was seeing

**Reply by ravindk89** [reply]
We're working backwards here.

What version of MinIO were you on when this *was* working?

What version of BOTO3 are you using?

What is the code you are firing to create the POST Presigned URL?
And what does the payload look like?

We need that information first.

> This was working fine until a recent release of MinIO

If it was working before and now it's not, it's not likely intentional. Let's do a bit of debugging :)
